src/toggle/toggle.tsx

```diff
@@ -0,0 +1,163 @@
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        setSelected(checked);
+    }, [checked]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleOnChange = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (!disabled) {
+                switch (type) {
+                    case "checkbox":
+                        {
+                            if (!selected) {
+                                setSelected((prevSelected) => {
+                                    return !prevSelected;
+                                });
+                            }
+                        }
+                        break;
+                    case "radio":
+                    case "yes":
+                    case "no":
+                        {
+                            if (!selected) {
+                                setSelected(true);
+                            }
+                        }
+                        break;
+                }
+
+                if (onChange) onChange(event);
+            }
+        },
+        []
+    );
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderIndicator = () => {
+        let toggleIconType: ToggleIconType;
+
+        switch (type) {
+            case "yes":
+                toggleIconType = "tick";
+                break;
+            case "no":
+                toggleIconType = "cross";
+                break;
+            case "checkbox":
+            case "radio":
+                toggleIconType = type;
+                break;
+        }
+
+        return (
+            <ToggleIcon
+                type={toggleIconType}
+                active={selected}
+                disabled={disabled}
+            />
+        );
+    };
+
+    // TODO: Reinstate later
+    // const renderSubLabel = () => {
+    //     if (!subLabel) {
+    //         return null;
+    //     }
+
+    //     let component: string | JSX.Element;
+
+    //     if (typeof subLabel === "string") {
+    //         component = subLabel;
+    //     } else if (typeof subLabel === "function") {
+    //         component = subLabel();
+    //     }
+
+    //     return (
+    //         <SubLabel disabled={disabled} $selected={selected}>
+    //             {component}
+    //         </SubLabel>
+    //     );
+    // };
+
+    return (
+        <Container
+            $selected={selected}
+            $disabled={disabled}
+            role={type === "checkbox" ? "checkbox" : "radio"}
```

Comment:
can remove this `role` attribute as well, it's causing an issue with Voiceover in Safari